infra/aws/terraform/prow-build-cluster/node_group_us-east-2a.tf

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+locals {
+  node_group_build_us_east_2a = {
+    name            = "build-us-east-2a"
+    description     = "EKS managed node group in US East 2a"
+    use_name_prefix = true
+
+    cluster_version = var.node_group_version_us_east_2a
+
+    taints = var.node_taints_build
```

Comment:
FYI - you can use [`eks_managed_node_group_defaults`](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/2cb1fac31b0fc2dd6a236b0c0678df75819c5a3b/examples/eks_managed_node_group/main.tf#L84)  to set the default values you wish to use across all of the managed nodegroups, and then any specific settings unique to that nodegroup can be set in the nodegroup definition. You can also set the defaults and still override the value in the nodegroup definition as well

So roughly speaking, something like the following might help cut down on the copy+pasta across nodegroup definitions:

```hcl
  eks_managed_node_group_defaults = {
    use_name_prefix = true
    
    taints = var.node_taints_build
    labels = var.node_labels_build
    ... anything else that you want common across the nodegroups
  }
  
  
```

Reply:
That's neat, thanks for pointing it out! I'll keep it in mind and see if we can make use of it the next time we do some rollout